examples/ssr-islands-axum/src/app.rs

```diff
@@ -94,11 +94,12 @@ mod header {
 
     #[component]
     pub fn View() -> impl IntoView {
+        let i18n = expect_i18n();
         view! {
             <header>
-                <A href="/">{move_tr!("home")}</A>
-                <LanguageSelector/>
-                <A href="/page-2">{move_tr!("page-2")}</A>
+                <A href="/">{move_tr!(i18n, "home")}</A>
+                <LanguageSelector />
+                <A href="/page-2">{move_tr!(i18n, "page-2")}</A>
```

Comment:
> * In the `header` module, I added some navigation links along with the `LanguageSelector`, as most users would typically place these in the header. The links don’t need to be inside an island, since they can inherit the i18n context from `App()`. To test behavior, I placed one link before the `LanguageSelector` and one after. I expected both links to be translated, but only the first one is. The `LanguageSelector` loads its own context, which I assumed would remain confined to the island, without affecting translations from the `#[component]` running on the server.

This is another case of weirdo behaviour of `provide_context` / `use_context` from Leptos. In v0.6, contexts are herratically spread from child components in a declarative way, but v0.7 "improves" (?) it by executing the child first and spreading the last to their parents. In v0.6, the first string is translated but in v0.7, none of them are translated.

I'm in the team that components should be always declared using `<Context/>` components to avoid these kinds of strange behaviours. Even I would remove `use_context` at all in the future from Leptos itself, looks like the root of all evils. In leptos-fluent v0.2 you'll be forced to pass a `children` prop `leptos_fluent!` because internally it will use `<Context/>`.

To specify exactly what i18n context to use, you can pass it as the first value to `move_tr!` and `tr!` macros.